HippoMocks/hippomocks.h

```diff
@@ -1256,6 +1263,57 @@ std::pair<int, int> virtual_index(T t)
 
 	if (conv.u.vindex != 0)
 		return std::pair<int, int>((conv.u.delta + conv.u.vtordisp)/sizeof(void*), conv.u.vindex * 2 + 1);
+	
+#elif defined (__SUNPRO_CC)
+    union {
+        T t;
+        struct
+        {
+            unsigned char *value;
+            unsigned long baseoffs;
+        } u;
+    } conv;
+    conv.t = t;
+   //is virtual call
+    unsigned char prolog[]= {0x55, 0x8b, 0xec, 0x83, 0xec,
```

Comment:
Feels sensitive to optimization or change... I dislike this code, but I also know we have the same for MSVC.